pkg/access_token/access_token_file_test.go

```diff
@@ -19,138 +19,223 @@ func TestAccessTokenFile(t *testing.T) {
+				tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite1"
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Finishes without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Checks that the file exists in the path defined", func (){
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite1")
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("When running Read method", func() {
+					dataExpected, _ := tokenInFile.Read()
+
+					// Confirm data was written
+					Convey("Returns the data the was written to the file", func() {
+						eq := reflect.DeepEqual(dataActual, dataExpected)
+						So(eq, ShouldEqual, true)
+					})
+				})
+			})
 		})
 
-		Convey("Returns error if Data input is nil", func() {
-			err := tokenInFile.Write(nil)
+		Convey("Given an access token without data", func() {
+
+			Convey("When running the Write method", func() {
+				err := tokenInFile.Write(nil)
 
-			So(err.Error(), ShouldEqual, "error writing access token, reason: data is empty")
+				Convey("Raises an error that the access token data is empty", func() {
+					So(err.Error(), ShouldEqual, "error writing access token, reason: data is empty")
+				})
+			})
 		})
 
-		Convey("Returns no error if file already exists", func() {
+		Convey("Given access token with data and a file location defined", func() {
 			dataActual := []byte{'t', 'e', 's', 't'}
 			tokenInFile.TokenFilePath = "/tmp/accessTokenFileWrite2"
-			err := tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
-
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
-
-			// Second time
-			dataActual = []byte{'t', 'e', 's', 't', '2'}
-			err = tokenInFile.Write(dataActual)
-			So(err, ShouldEqual, nil)
 
-			// Check if file exits
-			_, err = os.Stat("/tmp/accessTokenFileWrite1")
-			So(err, ShouldEqual, nil)
-
-			// Test we are reading the new Data
-			dataExpected, err := tokenInFile.Read()
-			eq := reflect.DeepEqual(dataActual, dataExpected)
-
-			So(err, ShouldEqual, nil)
-			So(eq, ShouldEqual, true)
+			Convey("When running the Write method", func() {
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Writes the access token without raising an error", func() {
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Creates with file without raising an error", func() {
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite2")
+					So(err, ShouldEqual, nil)
+				})
+			})
+
+			Convey("When running the Write method a second time", func() {
+				dataActual = []byte{'t', 'e', 's', 't', '2'}
+				err := tokenInFile.Write(dataActual)
+
+				Convey("Creates with file without raising an error", func() {
+					// Check if file exists
+					_, err = os.Stat("/tmp/accessTokenFileWrite2")
+					So(err, ShouldEqual, nil)
+				})
+
+				Convey("Writes the data the file as expected", func() {
+					dataExpected, _ := tokenInFile.Read()
```

Comment:
i'd rather do this with `os` read

Reply:
why? we want to mock our `read` function not `os.read`

Reply:
yes but the test here is on `write` and the `read` part is given. i'd rather not fall on given stages because of our failures